src/Plugin/Condition/UserHasRole.php

```diff
@@ -38,12 +39,21 @@
 class UserHasRole extends RulesConditionBase {
 
   /**
-   * {@inheritdoc}
+   * Evaluate if user has role(s).
+   *
+   * @param \Drupal\user\UserInterface $account
+   *   The account to check.
+   * @param \Drupal\user\RoleInterface[] $roles
+   *   Array of user roles.
+   * @param string $operation
+   *   Either "AND": user has all of roles.
+   *   Or "OR": user has at least one of all roles.
+   *   Defaults to "AND".
+   *
+   * @return bool
+   *   TRUE if the account is blocked.
    */
-  public function evaluate() {
-    $account = $this->getContextValue('user');
-    $roles = $this->getContextValue('roles');
-    $operation = $this->getContext('operation')->hasContextValue() ? $this->getContextValue('operation') : 'AND';
+  protected function doEvaluate(UserInterface $account, array $roles, $operation = 'ALL') {
```

Comment:
Should be AND offcourse.

Now throwing an InvalidArgumentException as fallback in the switch-statement.
This is also used in \Drupal\rules\Plugin\RulesAction\DataConvert